signature/jws/conformance_test.go

```diff
@@ -0,0 +1,120 @@
+package jws
+
+import (
+	"crypto/elliptic"
+	"crypto/x509"
+	"reflect"
+	"sort"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/notaryproject/notation-core-go/signature"
+	"github.com/notaryproject/notation-core-go/testhelper"
+)
+
+var (
+	// signedAttributes for signing request
+	signedAttributes = signature.SignedAttributes{
+		SigningScheme: signature.SigningSchemeX509,
+		SigningTime:   time.Now().Truncate(time.Second),
+		Expiry:        time.Now().Truncate(time.Second).Add(time.Hour * 24),
+		ExtendedAttributes: sortAttributes([]signature.Attribute{
+			{Key: "signedCritKey1", Value: "signedCritValue1", Critical: true},
+			{Key: "signedKey1", Value: "signedValue1", Critical: false},
+			{Key: "signedKey2", Value: "signedValue1", Critical: false},
+			{Key: "signedKey3", Value: "signedValue1", Critical: false},
+			{Key: "signedKey4", Value: "signedValue1", Critical: false},
+		}),
+	}
+	// unsignedAttributes for signing request
+	unsignedAttributes = signature.UnsignedAttributes{
+		SigningAgent: "NotationConformanceTest/1.0.0",
+	}
+	// payload to be signed
+	payload = signature.Payload{
+		ContentType: signature.MediaTypePayloadV1,
+		Content:     []byte("hello JWS"),
+	}
+	// certificate chain for signer
+	leafCertTuple = testhelper.GetECCertTuple(elliptic.P256())
+	certs         = []*x509.Certificate{leafCertTuple.Cert, testhelper.GetECRootCertificate().Cert}
+)
+
+func conformanceTestSignReq() *signature.SignRequest {
+	signer, err := signature.NewLocalSigner(certs, leafCertTuple.PrivateKey)
+	if err != nil {
+		panic(err)
+	}
+
+	return &signature.SignRequest{
+		Payload:                  payload,
+		Signer:                   signer,
+		SigningTime:              signedAttributes.SigningTime,
+		Expiry:                   signedAttributes.Expiry,
+		ExtendedSignedAttributes: signedAttributes.ExtendedAttributes,
+		SigningAgent:             unsignedAttributes.SigningAgent,
+		SigningScheme:            signedAttributes.SigningScheme,
+	}
+}
+
+// TestConformance generates JWS encoded message, parses the encoded message and
+// verify the payload, signed/unsigned attributes conformance.
+func TestConformance(t *testing.T) {
+	env := envelope{}
+	signReq := conformanceTestSignReq()
+	encoded, err := env.Sign(signReq)
```

Comment:
For conformance test, we should also check the payload and the protected headers in the envelope are as expected after the Sign process.

Reply:
Added TestSignedMessageConformance()